x-pack/legacy/plugins/alerting/server/extend_route_with_license_check.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { LicenseState, verifyApiAccessFactory } from './lib/license_state';
+
+export function extendRouteWithLicenseCheck(route: any, licenseState: LicenseState) {
+  const verifyApiAccessPreRouting = verifyApiAccessFactory(licenseState);
+
+  if (route.options) {
```

Comment:
Sorry for repeating a point, just pointing out this can be cleaned up much like the `extend_route_with_license_check.ts` under `actions`.

Also, and I know this might be a Kibana thing, but  it actually looks like these two components are identical - is there any way to share a component between Alerting and Actions rather than repeat code we need to maintain? 🤔